File: C++/문제_10819번.cpp
//문제 10819번 차이를 최대로 -> 백트래킹
#include <vector>
#include <cmath>
vector<int>v;
int n;
int num[9];
int result = 0;
bool visit[9];
void dfs(int cnt){
    if(cnt == n){
        int sum = 0;
        for(int i = 0;i<v.size() - 1;i++){
            sum += abs(v[i] - v[i+1]);
        }
        result = max(result, sum);
        return;
    }
    for(int i = 0;i<n;i++){
        if(visit[i] == 0){
            v.push_back(num[i]);
            visit[i] = 1;
            dfs(cnt + 1);
            visit[i] = 0;
            v.pop_back();
        }
    }
}
int main(){
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);
    cout.tie(NULL);
    cin >> n;
    for(int i = 0;i<n;i++){
        cin >> num[i];
    }
    dfs(0);
    cout << result;
}

File: JavaScript/문제_10819번.js
let input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");
let n = +input[0];
let arr = input[1].split(' ').map(o => +o);
let node = [];
let visit = new Array(n).fill(0);
let result = 0;
const dfs = (cnt) => {
  if (cnt == n) {
    let sum = 0;
    for (let i = 0; i < n - 1; i++) {
      sum += Math.abs(node[i] - node[i + 1]);
    }
    result = Math.max(result, sum);
    return;
  }
  for (let i = 0; i < n; i++) {
    if (visit[i] == 0) {
      visit[i] = 1;
      node.push(arr[i]);
      dfs(cnt + 1);
      node.pop();
      visit[i] = 0;
    }
  }
}
dfs(0);
console.log(result);